include/vcpkg/binarycaching.h

```diff
@@ -77,12 +78,15 @@ namespace vcpkg
 
         LocalizedString valid();
         std::string instantiate_variables(const Dependencies::InstallPlanAction& action) const;
+        std::string get_url(BinaryConfigParserState state);
     };
 
     struct BinaryConfigParserState
     {
         bool m_cleared = false;
         bool interactive = false;
+        std::set<std::string> binary_cache_providers;
```

Comment:
```suggestion
        std::set<StringLiteral> binary_cache_providers;
```

Does this work? I can't imagine that these wouldn't ever be fixed values.